src/Components/test/testassets/BasicTestApp/Program.cs

```diff
@@ -38,12 +39,16 @@ public static async Task Main(string[] args)
                     policy.RequireAssertion(ctx => ctx.User.Identity.Name?.StartsWith("B") ?? false));
             });
 
+            builder.Services.AddDataProtection();
+
+            builder.Services.AddTransient<ProtectedLocalStorage>();
```

Comment:
Why not use the extension method?

Reply:
For the test app, I used `AddTransient` so I could lazily instantiate the `ProtectedBrowserStorage` services. This allowed me to write tests to ensure `ProtectedBrowserStorage` only runs on Blazor server and throws an exception on Wasm.